relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -390,3 +402,44 @@ func (ccp *CosmosChainProcessor) queryCycle(ctx context.Context, persistence *qu
 
 	return nil
 }
+
+func (ccp *CosmosChainProcessor) CollectMetrics(ctx context.Context, persistence *queryCyclePersistence) {
+	ccp.CurrentBlockHeight(ctx, persistence)
+	ccp.CurrentRelayerBalance(ctx)
```

Comment:
It seems unnecessarily expensive to make this query every single queryCycle, which occurs roughly every second. The current block height is fine to update that frequently since it doesn't involve a query. Maybe we could have a time-since-last check here and do this one maybe once a minute?

Reply:
Maybe we do once every 10-20 blocks @joeabbey ?

Reply:
tuned to every 30 seconds to split the difference for "normal speed" chains like Osmosis and Cosmos... and fast-paced chains like Evmos